### DISCUSSION TITLE
GraphiQL stopped working

### DISCUSSION THREAD

**Discussion by aeither** [aptos-labs/aptos-developer-discussions#514] [Questions] [answered]
### Describe your question in detail.

This query was working, then it stopped working one day. I didn't change anything. why, what could be the issue?

https://cloud.hasura.io/public/graphiql?endpoint=https://api.testnet.aptoslabs.com/v1/graphql

```js
  query GetCollectionNfts($collection_name: String) {
    current_token_datas_v2(
      where: {
        current_collection: {
          collection_name: { _eq: $collection_name }
        }
      }
      order_by: { last_transaction_timestamp: desc }
    ) {
      token_name
      description
      token_uri
      collection_id
      last_transaction_timestamp
      token_data_id
      token_properties
      current_token_ownerships {
        owner_address
        amount
      }
    }
  }
```

```js
{
  "collection_name": "Open Forge - 8c68bc"
}
```

### What error, if any, are you getting?

{
  "errors": [
    {
      "message": "database query error",
      "extensions": {
        "path": "$",
        "code": "unexpected"
      }
    }
  ]
}

### Which operating system are you using?

macOS

### Which SDK or tool are you using? (if any)

Indexer API

**Comment by 0xmaayan** [comment, marked as answer]
Hi @aeither 

This is an expensive query, `last_transaction_timestamp` is not indexed and this will be very expensive for large collections. It's also expensive because you are filtering on `collection_name` which doesn't have an index. You should filter on `collection_id`

As for why it is suddenly errors out, it could be related to that we lowered the query time out in testnet: https://github.com/aptos-labs/internal-ops/pull/4400 - This is a change we deployed last week

**Reply by larry-aptos** [reply]
if you want to know more about collection id, you can check an example here: https://aptos.dev/en/build/indexer/token-metadata

**Reply by aeither** [reply]
> if you want to know more about collection id, you can check an example here: https://aptos.dev/en/build/indexer/token-metadata

How can I get the collection ID?

**Comment by larry-aptos** [comment]
There is a python code snippet to show you how to get collection id
